dotCMS/src/enterprise/java/com/dotcms/enterprise/publishing/remote/handler/ContentHandler.java

```diff
@@ -1049,32 +1051,48 @@ private String getUniqueMatchErrorMsg(final List<Field> uniqueFields, final Stri
                 matchedContent.getInode(), fieldsInfo.toString());
     }
 
-    /**
-     * Associates a list of tags coming from the bundle to the specified local content.
-     * 
-     * @param content - The {@link Contentlet} that will have the updated tags from the bundle.
-     * @param tagsFromSender - The list of {@link Tag} objects coming from the sender,
-     * @throws DotDataException Tags could not be read or saved to the data source.
-     */
-	private void relateTagsToContent(Contentlet content, Map<String, List<Tag>> tagsFromSender) throws DotDataException {
-		if(tagsFromSender!=null) {
-			for (Map.Entry<String, List<Tag>> fieldTags : tagsFromSender.entrySet()) {
-				String fieldVarName = fieldTags.getKey();
+	/**
+	 * Associates a list of tags coming from the bundle to the specified local content.
+	 *
+	 * @param content - The {@link Contentlet} that will have the updated tags from the bundle.
+	 * @param tagsFromSender - The list of {@link Tag} objects coming from the sender,
+	 * @throws DotDataException Tags could not be read or saved to the data source.
+	 */
+	@VisibleForTesting
+	void relateTagsToContent(Contentlet content, Map<String, List<Tag>> tagsFromSender) throws DotDataException {
+		if(tagsFromSender==null || tagsFromSender.isEmpty()) {
+			return;
+		}
 
-				for (Tag remoteTag : fieldTags.getValue()) {
-					Tag localTag = tagAPI.getTagByNameAndHost(remoteTag.getTagName(), remoteTag.getHostId());
+		for (Map.Entry<String, List<Tag>> fieldTags : tagsFromSender.entrySet()) {
+			String fieldVarName = fieldTags.getKey();
 
-					// if there is NO local tag, save the one coming from remote, otherwise use local
-					if (localTag == null || Strings.isNullOrEmpty(localTag.getTagId())) {
-						localTag = tagAPI.saveTag(remoteTag.getTagName(), remoteTag.getUserId(), remoteTag.getHostId());
-					}
+			for (Tag remoteTag : fieldTags.getValue()) {
+				Tag localTag = tagAPI.getTagByNameAndHost(remoteTag.getTagName(), remoteTag.getHostId());
 
-					TagInode localTagInode = tagAPI.getTagInode(localTag.getTagId(), content.getInode(), fieldVarName);
+				String localUserId = Try.of(()->APILocator.getUserAPI().loadUserById(remoteTag.getUserId()).getUserId()).getOrElse(APILocator.systemUser().getUserId());
 
-					// avoid relating tags twice
-					if(localTagInode==null || !Strings.isNullOrEmpty(localTagInode.getTagId())) {
-						tagAPI.addContentletTagInode(localTag, content.getInode(), fieldVarName);
-					}
+				Host tagSite = Try.of(()->APILocator.getHostAPI().find(remoteTag.getHostId(), APILocator.systemUser(), false)).getOrNull();
+				Host contentSite = Try.of(()->APILocator.getHostAPI().find(content.getIdentifier(), APILocator.systemUser(), false)).getOrNull();
+
+				final String localSiteId = UtilMethods.isSet(()->tagSite.getTagStorage())
+						? tagSite.getTagStorage()
+						: UtilMethods.isSet(()->contentSite.getTagStorage())
+								? contentSite.getTagStorage()
+								: Host.SYSTEM_HOST;
```

Comment:
Should we also log a message indicating that the Tag is being saved to System Host? Something similar to what was done when pushing Content Types living in a Site that doesn't exist in the receiver.

Reply:
unlike a piece of content, saving a tag to the system host does not really affect permissions and is not a huge deal.  We can, I just did not want to get too noisy.